mmv1/third_party/terraform/services/netapp/resource_netapp_volume_test.go

```diff
@@ -0,0 +1,214 @@
+			},
+		},
+	})
+}
+
+func testAccNetappvolume_volumeCreateExample_full(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_netapp_storage_pool" "default" {
+  name = "tf-test-test-pool%{random_suffix}"
+  location = "us-central1"
+  service_level = "PREMIUM"
+  capacity_gib = "2048"
+  network = data.google_compute_network.default.id
+}
+
+
+resource "google_netapp_volume" "test_volume" {
+  location = "us-central1"
+  name = "tf-test-test-volume%{random_suffix}"
+  capacity_gib = "100"
+  share_name = "testshare"
+  storage_pool = google_netapp_storage_pool.default.name
+  protocols = ["NFSV3"]
+  smb_settings = []
+  unix_permissions = "0770"
+  labels = {
+	key= "test"
+    value= "pool"
+  }
+  description = "This is a test description"
+  snapshot_directory = false
+  security_style = "UNIX"
+  kerberos_enabled = false
+  export_policy {
+    rules {
+		access_type           = "READ_WRITE"
+		allowed_clients       = "0.0.0.0/0"
+		has_root_access       = "false"
+		kerberos5_read_only   = false
+		kerberos5_read_write  = false
+		kerberos5i_read_only  = false
+		kerberos5i_read_write = false
+		kerberos5p_read_only  = false
+		kerberos5p_read_write = false
+		nfsv3                 = true
+		nfsv4                 = false
+    }
+  }
+  restricted_actions = []
+  snapshot_policy {
+    daily_schedule {
+      hour              = 1
+      minute            = 1
+      snapshots_to_keep = 2
+    }
+    enabled = true
+    hourly_schedule {
+      minute            = 20
+      snapshots_to_keep = 2
+    }
+    monthly_schedule {
+      days_of_month     = "5"
+      hour              = 2
+      minute            = 3
+      snapshots_to_keep = 4
+    }
+    weekly_schedule {
+      day               = "Wednesday"
+      hour              = 2
+      minute            = 3
+      snapshots_to_keep = 2
+    }
+  }
+
+}
+
+data "google_compute_network" "default" {
+  name = "%{network_name}"
+}
+`, context)
+}
+func testAccNetappvolume_volumeCreateExample_update(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_netapp_storage_pool" "default" {
+  name = "tf-test-test-pool%{random_suffix}"
+  location = "us-central1"
+  service_level = "PREMIUM"
+  capacity_gib = "2048"
+  network = data.google_compute_network.default.id
+}
+
+
+resource "google_netapp_volume" "test_volume" {
+  location = "us-central1"
+  name = "tf-test-test-volume%{random_suffix}"
+  capacity_gib = "200"
+  share_name = "testshare"
+  storage_pool = google_netapp_storage_pool.default.name
+  protocols = ["NFSV3"]
+  smb_settings = []
+  unix_permissions = "0770"
+  labels = {}
+  description = ""
+  snapshot_directory = false
+  security_style = "UNIX"
+  kerberos_enabled = false
+  export_policy {
+    rules {
+		access_type           = "READ_ONLY"
+		allowed_clients       = "0.0.0.0/0"
+		has_root_access       = "true"
+		kerberos5_read_only   = false
```

Comment:
Can we change the value of these booleans between the update steps?

Reply:
has_root_access can be changed easily. All the kerberos parameters can only be changed it the volumes would be exported with Kerberos and that requires complex infrastructure to be in place in the test environment. Not realistic.